### PR TITLE
Add methods to copy snapshots

### PR DESCRIPTION
**Pull request by pabloferz**

### CODE REVIEW

**Comment by InnocentBug** [comment]
Is this ready for review, or do you want to add something.
I think we said the last time that something was missing here.

**Comment by pabloferz** [comment]
Yes, I wanted to add an option to indicate if we want the copy on the CPU if it's on the GPU.

**Comment by pabloferz** [comment]
All right. This is ready for review.

File: pysages/backends/snapshot.py [hunk 1]
```diff
@@ -2,25 +2,49 @@
...
+from pysages.utils import copy
 
 import jax.numpy as np
 
-from collections import namedtuple
 
-
-class Box(namedtuple("Box", ("H", "origin"))):
+class Box(NamedTuple("Box", [
+    ("H",      JaxArray),
```
Comment (InnocentBug) [inline pysages/backends/snapshot.py:14]: Is "H" really good name for this.
I haven't seen it elsewhere and could we use a better telling name?

File: tests/test_snapshots.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    old_ptr = old.positions.unsafe_buffer_pointer()
+    new_ptr = new.positions.unsafe_buffer_pointer()
+    old_box_H_ptr = old.box.H.unsafe_buffer_pointer()
+    new_box_H_ptr = new.box.H.unsafe_buffer_pointer()
+
+    # When copying to CPU we get a `numpy.ndarray` instead of a
+    # `jaxlib.xla_extension.DeviceArray`
+    new_cpu = copy(old, ToCPU())
+
+    assert np.all(old.positions == new.positions).item()
```
Comment (InnocentBug) [inline tests/test_snapshots.py:35]: Ok, I was double checking if assert is the right way to test in automated frameworks.
But it seems like that pytest relies on assert, while unittest wants exceptions.

**Review by InnocentBug — APPROVED** [review]
Looks good to me.

I am happy that we have a test case now :)

File: pysages/backends/snapshot.py (hunk 1 above)
Comment (pabloferz) [reply pysages/backends/snapshot.py:14]: Agreed, I'm not sure what to call it, though. It represents the affine transform that needs to be applied to the positions for not orthogonal boxes. Maybe `transform` is good enough?

File: tests/test_snapshots.py (hunk 2 above)
Comment (pabloferz) [reply tests/test_snapshots.py:35]: Do you have a preference on `unittest`. I like that `pytest` doesn't require too much boilerplate.

Comment (InnocentBug) [reply tests/test_snapshots.py:35]: no, I don't have a preference. My testing experience is more about C++. I read up on it and seems like it is as you say. both are fine and pytest is easier to work with. So I am happy to go along with it.

File: pysages/backends/snapshot.py (hunk 1 above)
Comment (InnocentBug) [reply pysages/backends/snapshot.py:14]: `transform` sounds good to me

Comment (pabloferz) [reply pysages/backends/snapshot.py:14]: OK. Will change that in a separate PR. Merging this for now.

**Pull request merged**